include/highfive/bits/H5Node_traits.hpp

```diff
@@ -110,7 +110,38 @@ class NodeTraits {
     ///
     /// \brief return the name of the object with the given index
     /// \return the name of the object
-    std::string getObjectName(size_t index) const;
+    std::string getName(size_t index) const;
+
+    ///
+    /// \brief return the path to the current object
+    /// \return the path to the object
+    std::string getPath() const;
```

Comment:
By placing `getPath` in Object you will have it in `DataSet`, `DataSpace`, `Group`, Attribute`. Is there still functionality missing? If not, I think this should be removed.